core/domain/feature_gating_services.py

```diff
@@ -0,0 +1,177 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services for platform parameters."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from constants import constants
+from core import features_registry
+from core.domain import platform_parameter_domain as param_domain
+from core.domain import platform_parameter_registry as registry
+
+
+ALL_FEATURES_LIST = (
+    features_registry.DEV_FEATURES_LIST +
+    features_registry.TEST_FEATURES_LIST +
+    features_registry.PROD_FEATURES_LIST
+)
+
+ALL_FEATURES_NAMES_SET = set(ALL_FEATURES_LIST)
+
+
+def create_evaluation_context_for_client(client_context_dict):
+    """Returns context instance for evaluation, using the information
+    provided by clients.
+
+    Args:
+        client_context_dict: dict. The client side context.
+
+    Returns:
+        EvaluationContext. The context for evaluation.
+    """
+    return param_domain.EvaluationContext.from_dict(
+        client_context_dict,
+        {
+            'server_mode': _get_running_mode()
+        }
+    )
+
+
+def get_all_feature_flag_setting_dicts():
```

Comment:
Why not just ``get_all_feature_flag_dicts``?

And, why not return the parameters themselves rather than the dicts? How are you planning to use this?

Reply:
Per my second question above, can you also provide context about how you plan to use this function? That's mainly for my benefit so that I can review this from the perspective of how it should be used per its API contract.

Reply:
They are used for providing the admin panel with the feature flag information, so that on the admin panel the administrator can see what features are there and what the rules look like.
Also added brief description to the docstrings.

Reply:
Got it--thanks!